src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1017,13 +1016,45 @@ class CephExporterService(CephService):
 
     def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonDeploySpec:
         assert self.TYPE == daemon_spec.daemon_type
+        spec = cast(CephExporterSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
+        keyring = self.get_keyring_with_caps(self.get_auth_entity(daemon_spec.daemon_id),
+                                             ['mon', 'profile ceph-exporter',
+                                              'mon', 'allow *',
+                                              'mgr', 'allow rw',
+                                              'osd', 'allow rwx'])
+
+        if spec.sock_dir:
+            ret, out, err = self.mgr.check_mon_command({
+                'prefix': 'config set',
+                'who': utils.name_to_config_section(daemon_spec.name()),
+                'name': 'exporter_sock_dir',
+                'value': spec.sock_dir,
+                })                   
+        if spec.port:
+            ret, out, err = self.mgr.check_mon_command({
+                'prefix': 'config set',
+                'who': utils.name_to_config_section(daemon_spec.name()),
+                'name': 'exporter_port',
+                'value': spec.port,
+            })
+        if spec.prio_limit:
+            ret, out, err = self.mgr.check_mon_command({
```

Comment:
Not sure what's wrong here but facing this issue:
`config set failed: bad or missing field 'value' retval: -22`
But it works fine from cli:
`$ ceph config set <daemon_name> exporter_prio_limit 5`
>> 5`
@adk3798 @rkachach Any idea on this?